GraphQL+OIDC issue with default configuration and native

### Describe the bug

In Quarkus 3.30 it was possible to use a configuration like this:
```
%option1.quarkus.oidc-client.auth-server-url=http://${KCHOST:host.docker.internal}:8543/realms/quarkus
%option1.quarkus.oidc-client.client-id=quarkus-app
%option1.quarkus.oidc-client.credentials.secret=secret
%option1.quarkus.oidc-client.grant.type=password
%option1.quarkus.oidc-client.grant-options.password.username=alice
%option1.quarkus.oidc-client.grant-options.password.password=alice
```

and configure a graphql client to use:
```
@GraphQLClientApi(configKey = "foo_bar")
public interface MyTypesafeClientAPI {
```

this is not working anymore starting in 3.31.
there are actually 2 issues:
- the default configuration is not supported anymore
- even when workarounding this issue, there is another issue when trying to work in native

Those issues have been introduced in Quarkus 3.31. 
And the issues have been tested to remain in 3.33 up to 3.37.
See https://github.com/quarkusio/quarkus/pull/51743 and https://github.com/quarkusio/quarkus/pull/51764

cc @michalvavrik 

### Expected behavior

the default configuration should be accepted
this should work in native

### Actual behavior

the default configuration will lead to an exception when trying to start the app in jvm mode:
```
2026-07-27 18:38:44,380 ERROR [io.quarkus.runtime.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:112)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:129)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:143)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:70)
        at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:36)
Caused by: java.lang.NullPointerException: Cannot invoke "String.startsWith(String)" because "name" is null
        at io.quarkus.commons.classloading.ClassLoaderHelper.isInJdkPackage(ClassLoaderHelper.java:44)
        at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:94)
        at io.quarkus.bootstrap.runner.RunnerClassLoader.loadClass(RunnerClassLoader.java:86)
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.loadClass(OidcGraphQLClientIntegrationRecorder.java:33)
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.lambda$enhanceGraphQLClientConfigurationWithOidc$0(OidcGraphQLClientIntegrationRecorder.java:24)
        at java.base/java.util.HashMap.forEach(HashMap.java:1429)
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.enhanceGraphQLClientConfigurationWithOidc(OidcGraphQLClientIntegrationRecorder.java:18)
        at io.quarkus.runner.recorded.OidcGraphQLClientIntegrationProcessor$initialize1579852025.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.OidcGraphQLClientIntegrationProcessor$initialize1579852025.deploy(Unknown Source)
        ... 9 more

```

and when using the workaround `option2`, then we have the issue with native:
```
2026-07-27 16:48:25,529 ERROR [io.quarkus.runtime.Application] (main) Failed to start application: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
        at io.quarkus.runtime.Application.start(Application.java:116)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:119)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:79)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:50)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:143)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.RuntimeException: Failed to load generated class io.quarkus.oidc.client.graphql.runtime.AbstractGraphQLTokenProvider_foo_bar
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.loadClass(OidcGraphQLClientIntegrationRecorder.java:35)
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.lambda$enhanceGraphQLClientConfigurationWithOidc$0(OidcGraphQLClientIntegrationRecorder.java:24)
        at java.base@25.0.3/java.util.HashMap.forEach(HashMap.java:1430)
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.enhanceGraphQLClientConfigurationWithOidc(OidcGraphQLClientIntegrationRecorder.java:18)
        at io.quarkus.runner.recorded.OidcGraphQLClientIntegrationProcessor$initialize1579852025.deploy_0(Unknown Source)
        at io.quarkus.runner.recorded.OidcGraphQLClientIntegrationProcessor$initialize1579852025.deploy(Unknown Source)
        ... 7 more
Caused by: java.lang.ClassNotFoundException: io.quarkus.oidc.client.graphql.runtime.AbstractGraphQLTokenProvider_foo_bar
        at java.base@25.0.3/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:63)
        at java.base@25.0.3/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
        at io.quarkus.oidc.client.graphql.runtime.OidcGraphQLClientIntegrationRecorder.loadClass(OidcGraphQLClientIntegrationRecorder.java:33)
        ... 12 more
```

### How to Reproduce?

follow steps in readme of https://github.com/TechAndOperations/quarkus_reproducers/tree/issue_graphql_oidc

to workaround the native issue, one can add:
```
@RegisterForReflection(classNames = {"io.quarkus.oidc.client.graphql.runtime.AbstractGraphQLTokenProvider_foo_bar"})
public class ReflectionConfig {
}
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

21

### Quarkus version or git rev

`3.31.4` `3.33.2.1` `3.37.4`

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/oidc-client-graphql/deployment/src/main/java/io/quarkus/oidc/client/graphql/OidcGraphQLClientIntegrationProcessor.java
+++ b/extensions/oidc-client-graphql/deployment/src/main/java/io/quarkus/oidc/client/graphql/OidcGraphQLClientIntegrationProcessor.java
@@ -27,6 +27,7 @@ import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.annotations.Record;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 import io.quarkus.gizmo2.ClassOutput;
 import io.quarkus.gizmo2.Const;
 import io.quarkus.gizmo2.Gizmo;
@@ -100,6 +101,7 @@ class OidcGraphQLClientIntegrationProcessor {
      */
     @BuildStep
     void generateGraphQLTokenProducers(GraphQLTokenProducerInfo graphQLTokenProducerInfo,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassProducer,
             BuildProducer<GeneratedBeanBuildItem> generatedBeanProducer) {
         graphQLTokenProducerInfo.oidcClientToTokenProducerBeanName.forEach((oidcClientName, generatedClassName) -> {
             ClassOutput classOutput = new GeneratedBeanGizmo2Adaptor(generatedBeanProducer);
@@ -114,12 +116,17 @@ class OidcGraphQLClientIntegrationProcessor {
                     ctorCreator.public_();
                     ctorCreator
                             .body(bc -> {
+                                var oidcClientNameValue = DEFAULT_CLIENT_KEY.equals(oidcClientName) ? Const.ofNull(String.class)
+                                        : Const.of(oidcClientName);
                                 bc.invokeSpecial(ConstructorDesc.of(tokenProviderClassDesc, String.class), cc.this_(),
-                                        Const.of(oidcClientName));
+                                        oidcClientNameValue);
                                 bc.return_();
                             });
                 });
             });
+            reflectiveClassProducer.produce(ReflectiveClassBuildItem.builder(generatedClassName)
+                    .reason(getClass().getName())
+                    .publicConstructors().methods().fields().build());
         });
     }
 
@@ -143,7 +150,8 @@ class OidcGraphQLClientIntegrationProcessor {
         config.additionalOidcClients()
                 .forEach((graphQlClient, oidcClient) -> configKeysToOidcClients.put(graphQlClient, oidcClient.clientName()));
 
-        recorder.enhanceGraphQLClientConfigurationWithOidc(configKeysToOidcClients, config.clientName().orElse(null),
+        recorder.enhanceGraphQLClientConfigurationWithOidc(configKeysToOidcClients,
+                config.clientName().orElse(DEFAULT_CLIENT_KEY),
                 graphQLTokenProducerInfo.oidcClientToTokenProducerBeanName);
     }
 

--- a/extensions/oidc-client-graphql/runtime/src/main/java/io/quarkus/oidc/client/graphql/runtime/AbstractGraphQLTokenProvider.java
+++ b/extensions/oidc-client-graphql/runtime/src/main/java/io/quarkus/oidc/client/graphql/runtime/AbstractGraphQLTokenProvider.java
@@ -26,6 +26,6 @@ public abstract class AbstractGraphQLTokenProvider extends AbstractTokensProduce
 
     @Override
     protected Optional<String> clientId() {
-        return Optional.of(oidcClientId);
+        return Optional.ofNullable(oidcClientId);
     }
 }

--- a/integration-tests/oidc-dev-services/pom.xml
+++ b/integration-tests/oidc-dev-services/pom.xml
@@ -25,6 +25,14 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-oidc-client-graphql</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-graphql</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
             <artifactId>quarkus-websockets-next</artifactId>
         </dependency>
         <dependency>
@@ -83,6 +91,32 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-websockets-next-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-oidc-client-graphql-deployment</artifactId>
+            <version>${project.version}</version>
+            <type>pom</type>
+            <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-smallrye-graphql-deployment</artifactId>
             <version>${project.version}</version>
             <type>pom</type>
             <scope>test</scope>

--- a/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/GraphQLClientResource.java
+++ b/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/GraphQLClientResource.java
@@ -1,0 +1,27 @@
+package io.quarkus.it.oidc.dev.services;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+@Path("/oidc-graphql-test")
+public class GraphQLClientResource {
+
+    @Inject
+    TypesafeGraphQLClient defaultOidcClient;
+
+    @Inject
+    TypesafeGraphQLClientWithNamedOidcClient namedOidcClient;
+
+    @GET
+    @Path("/default")
+    public String callWithDefaultOidcClient() {
+        return defaultOidcClient.principalName();
+    }
+
+    @GET
+    @Path("/named")
+    public String callWithNamedOidcClient() {
+        return namedOidcClient.principalName();
+    }
+}

--- a/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/ProtectedGraphQLEndpoint.java
+++ b/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/ProtectedGraphQLEndpoint.java
@@ -1,0 +1,23 @@
+package io.quarkus.it.oidc.dev.services;
+
+import java.security.Principal;
+
+import jakarta.inject.Inject;
+
+import org.eclipse.microprofile.graphql.GraphQLApi;
+import org.eclipse.microprofile.graphql.Query;
+
+import io.quarkus.security.Authenticated;
+
+@GraphQLApi
+@Authenticated
+public class ProtectedGraphQLEndpoint {
+
+    @Inject
+    Principal principal;
+
+    @Query
+    public String principalName() {
+        return principal.getName();
+    }
+}

--- a/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/TypesafeGraphQLClient.java
+++ b/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/TypesafeGraphQLClient.java
@@ -1,0 +1,12 @@
+package io.quarkus.it.oidc.dev.services;
+
+import org.eclipse.microprofile.graphql.Query;
+
+import io.smallrye.graphql.client.typesafe.api.GraphQLClientApi;
+
+@GraphQLClientApi(configKey = "typesafe-oidc-client")
+public interface TypesafeGraphQLClient {
+
+    @Query
+    String principalName();
+}

--- a/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/TypesafeGraphQLClientWithNamedOidcClient.java
+++ b/integration-tests/oidc-dev-services/src/main/java/io/quarkus/it/oidc/dev/services/TypesafeGraphQLClientWithNamedOidcClient.java
@@ -1,0 +1,14 @@
+package io.quarkus.it.oidc.dev.services;
+
+import org.eclipse.microprofile.graphql.Query;
+
+import io.quarkus.oidc.client.filter.OidcClientFilter;
+import io.smallrye.graphql.client.typesafe.api.GraphQLClientApi;
+
+@GraphQLClientApi(configKey = "typesafe-named-oidc-client")
+@OidcClientFilter("graphql-oidc-client")
+public interface TypesafeGraphQLClientWithNamedOidcClient {
+
+    @Query
+    String principalName();
+}

--- a/integration-tests/oidc-dev-services/src/main/resources/application.properties
+++ b/integration-tests/oidc-dev-services/src/main/resources/application.properties
@@ -1,6 +1,23 @@
 quarkus.oidc.devservices.enabled=true
 quarkus.oidc.devservices.roles.Ronald=admin
 
+quarkus.oidc-client.auth-server-url=${quarkus.oidc.auth-server-url}
+quarkus.oidc-client.client-id=${quarkus.oidc.client-id}
+quarkus.oidc-client.credentials.secret=${quarkus.oidc.credentials.secret}
+quarkus.oidc-client.grant.type=password
+quarkus.oidc-client.grant-options.password.username=alice
+quarkus.oidc-client.grant-options.password.password=alice
+
+quarkus.oidc-client.graphql-oidc-client.auth-server-url=${quarkus.oidc.auth-server-url}
+quarkus.oidc-client.graphql-oidc-client.client-id=${quarkus.oidc.client-id}
+quarkus.oidc-client.graphql-oidc-client.credentials.secret=${quarkus.oidc.credentials.secret}
+quarkus.oidc-client.graphql-oidc-client.grant.type=password
+quarkus.oidc-client.graphql-oidc-client.grant-options.password.username=alice
+quarkus.oidc-client.graphql-oidc-client.grant-options.password.password=alice
+
+quarkus.smallrye-graphql-client.typesafe-oidc-client.url=http://localhost:${quarkus.http.test-port:8081}/graphql
+quarkus.smallrye-graphql-client.typesafe-named-oidc-client.url=http://localhost:${quarkus.http.test-port:8081}/graphql
+
 quarkus.websockets-next.server.supported-subprotocols=quarkus
 quarkus.websockets-next.server.propagate-subprotocol-headers=true
 

--- a/integration-tests/oidc-dev-services/src/test/java/io/quarkus/it/oidc/dev/services/OidcGraphQLClientIT.java
+++ b/integration-tests/oidc-dev-services/src/test/java/io/quarkus/it/oidc/dev/services/OidcGraphQLClientIT.java
@@ -1,0 +1,8 @@
+package io.quarkus.it.oidc.dev.services;
+
+import io.quarkus.test.junit.QuarkusIntegrationTest;
+
+@QuarkusIntegrationTest
+public class OidcGraphQLClientIT extends OidcGraphQLClientTest {
+
+}

--- a/integration-tests/oidc-dev-services/src/test/java/io/quarkus/it/oidc/dev/services/OidcGraphQLClientTest.java
+++ b/integration-tests/oidc-dev-services/src/test/java/io/quarkus/it/oidc/dev/services/OidcGraphQLClientTest.java
@@ -1,0 +1,28 @@
+package io.quarkus.it.oidc.dev.services;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+
+@QuarkusTest
+public class OidcGraphQLClientTest {
+
+    @Test
+    void defaultOidcClient() {
+        RestAssured.when().get("/oidc-graphql-test/default")
+                .then()
+                .statusCode(200)
+                .body(equalTo("alice"));
+    }
+
+    @Test
+    void namedOidcClient() {
+        RestAssured.when().get("/oidc-graphql-test/named")
+                .then()
+                .statusCode(200)
+                .body(equalTo("alice"));
+    }
+}

